src/main/java/uk/gov/hmcts/reform/unspec/handler/callback/DispatchBusinessProcessCallbackHandler.java

```diff
@@ -0,0 +1,55 @@
+package uk.gov.hmcts.reform.unspec.handler.callback;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.ccd.client.model.AboutToStartOrSubmitCallbackResponse;
+import uk.gov.hmcts.reform.ccd.client.model.CallbackResponse;
+import uk.gov.hmcts.reform.unspec.callback.Callback;
+import uk.gov.hmcts.reform.unspec.callback.CallbackHandler;
+import uk.gov.hmcts.reform.unspec.callback.CallbackParams;
+import uk.gov.hmcts.reform.unspec.callback.CallbackType;
+import uk.gov.hmcts.reform.unspec.callback.CaseEvent;
+import uk.gov.hmcts.reform.unspec.model.BusinessProcess;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import static uk.gov.hmcts.reform.unspec.callback.CaseEvent.DISPATCH_BUSINESS_PROCESS;
+import static uk.gov.hmcts.reform.unspec.model.BusinessProcessStatus.DISPATCHED;
+import static uk.gov.hmcts.reform.unspec.model.BusinessProcessStatus.READY;
+
+@Service
+@RequiredArgsConstructor
+public class DispatchBusinessProcessCallbackHandler extends CallbackHandler {
+
+    private static final List<CaseEvent> EVENTS = Collections.singletonList(DISPATCH_BUSINESS_PROCESS);
+
+    private final ObjectMapper mapper;
+
+    @Override
+    protected Map<CallbackType, Callback> callbacks() {
+        return Map.of(CallbackType.ABOUT_TO_SUBMIT, this::dispatchBusinessProcess);
+    }
+
+    @Override
+    public List<CaseEvent> handledEvents() {
+        return EVENTS;
+    }
+
+    private CallbackResponse dispatchBusinessProcess(CallbackParams callbackParams) {
+        Map<String, Object> data = callbackParams.getRequest().getCaseDetails().getData();
+        BusinessProcess businessProcess = mapper.convertValue(data.get("businessProcess"), BusinessProcess.class);
+        if (businessProcess.getStatus() == READY) {
+            data.put("businessProcess", BusinessProcess.builder()
+                .camundaEvent(businessProcess.getCamundaEvent())
+                .status(DISPATCHED)
+                .build());
```

Comment:
Is this also going to be resetting the other Business Process attributes? Would it be safer to explicitly set the action is and process instance id to null as well here?

Reply:
New object is created here, so anything not set will be null.